src/plugins/core/carousel.ts

```diff
@@ -122,19 +122,19 @@ export class CarouselPlugin extends CorePlugin<CarouselState> implements Carouse
 
   import(data: WorkbookData) {
     for (const sheet of data.sheets) {
-      const carousels = (sheet.figures || []).filter((figure) => figure.tag === "carousel");
-      for (const carousel of carousels) {
-        this.history.update("carousels", sheet.id, carousel.id, { items: carousel.data.items });
+      for (const carouselId in sheet.carousels || {}) {
+        const { figureId, carousel } = sheet.carousels[carouselId];
+        this.history.update("carousels", sheet.id, figureId, carousel);
       }
     }
   }
 
   export(data: WorkbookData) {
     for (const sheet of data.sheets) {
-      const carousels = sheet.figures.filter((figure) => figure.tag === "carousel");
-      for (const carousel of carousels) {
-        if (this.carousels[sheet.id]?.[carousel.id]) {
-          carousel.data = { ...carousel.data, ...this.carousels[sheet.id]?.[carousel.id] };
+      for (const carouselId in this.carousels[sheet.id] || {}) {
```

Comment:
Is it `carouselId` or `figureId`. `import` and `export` are not consistent. And I don't know which is which ?
```suggestion
      for (const figureId in this.carousels[sheet.id] || {}) {
```